tests/sfko/sfko/check.py

```diff
@@ -0,0 +1,211 @@
+    if len(succeeded_parts) != succeeded_parts[-1]:
+        passed = False
+    needed_copies = math.ceil(len(succeeded_parts) / 32)
+    if needed_copies > 1:
+        needed_copies += 1 # Add one for the final stage
+    succeeded_copies = len([x.extra['part'] for x in copys if x.status == 200])
+    if needed_copies != succeeded_copies:
+        passed = False
+    return passed
+
+@register_check('check-mpu')
+def check_mpu(bucket_conf, objs):
+    return False
+    if bucket_conf.backend.type is BackendType.AWS:
+        for bucket, key in objs.objects:
+            if not _check_aws_mpu(bucket, key, bucket_conf.backend.type.friendly()):
+                return False
+    elif bucket_conf.backend.type is BackendType.GCP:
+        for bucket, key in objs.objects:
+            if not _check_gcp_mpu(bucket, key, bucket_conf.backend.type.friendly()):
+                return False
+    return True
+
+@register_check('check-etag-mpu')
+def check_etag_mpu(bucket_conf, objs):
+    return False
+    for bucket, key in objs.objects:
+        delete_key, requests = start_after(
+                                    filter_requests(
+                                        get_requests(
+                                            bucket,
+                                            key,
+                                            bucket_conf.backend.type.friendly()),
+                                        method='HEAD'),
+                                    tag='mpu_comp')
+
+        for request in requests:
+            if bucket_conf.backend.type is BackendType.GCP:
+                header = 'x-goog-meta-scal-etag'
+                value = objs.etag
+            else:
+                header = 'ETag'
+                value = '"%s"'%objs.etag
+            if not _check_headers_for_request(request, {header: value}, required=False):
+                return False
+    return True
+
+def _wait_for_replication(bucket_conf, objs):
+    time.sleep(10)
+    expiry = 600 + time.time() # 10 Minutes from now
+    while expiry > time.time():
+        if count_pending_replication(bucket_conf.backend.type.friendly()) > 0:
+            time.sleep(5)
+        else:
+            _log.debug('Replication complete')
+            return True
+    _log.error('Some objects failed to replicate before the timeout elapsed!')
+    _log.debug(get_pending_replication(bucket_conf.backend.type.friendly()))
+    return False
+
+
+@register_check('check-replication')
+def check_replication(bucket_conf, objs):
+    if _wait_for_replication(bucket_conf, objs):
+        for bucket, key in objs.objects:
+            if not _check_backend_key(bucket, key, bucket_conf.replication.type.friendly()):
+                return False
+        return True
+    return False
+
+
+@register_check('check-replication-mpu')
+def check_replication_mpu(bucket_conf, objs):
+    if _wait_for_replication(bucket_conf, objs):
+        if bucket_conf.replication.type is BackendType.AWS:
```

Comment:
This block seems to do what `check_mpu()` already does, could call the former maybe

Reply:
The difference being, `check_replication_mpu` references  `bucket_conf.replication` instead of `bucket_conf.backend`. These con probably be combined in the future, but I want to see how the check functions evolve before aggressively merging them